### DISCUSSION TITLE
Get width and height from SPS and PPS

### DISCUSSION THREAD

**Discussion by cedricve** [bluenviron/gortsplib#475] [Questions] [answered]
Hey there,

I was wondering how to calculate the width and height from the PPS and SPS (h264). Couldn't find an example, I would like to be able to know without decoding an image.

Kind regards
Cedric

**Comment by aler9** [comment, marked as answer]
Hello, i remember i posted an example some time ago but i can't find it anymore, anyway, here's what you're looking for.

```go
package main

import (
	"fmt"

	"github.com/bluenviron/gortsplib/v4"
	"github.com/bluenviron/gortsplib/v4/pkg/base"
	"github.com/bluenviron/gortsplib/v4/pkg/format"
	"github.com/bluenviron/mediacommon/pkg/codecs/h264"
)

func main() {
	c := gortsplib.Client{}

	// parse URL
	u, err := base.ParseURL("rtsp://localhost:8554/mystream")
	if err != nil {
		panic(err)
	}

	// connect to the server
	err = c.Start(u.Scheme, u.Host)
	if err != nil {
		panic(err)
	}
	defer c.Close()

	// find available medias
	desc, _, err := c.Describe(u)
	if err != nil {
		panic(err)
	}

	// find the H264 media and format
	var forma *format.H264
	medi := desc.FindFormat(&forma)
	if medi == nil {
		panic("media not found")
	}

	var sps h264.SPS
	err = sps.Unmarshal(forma.SPS)
	if err != nil {
		panic(err)
	}

	fmt.Println("Width:", sps.Width())
	fmt.Println("Height:", sps.Height())
}
```

**Reply by cedricve** [reply]
I just found some example code, but didn't find it. You are god! <3

    self.Width = (self.MbWidth * 16) - self.CropLeft*2 - self.CropRight*2
    self.Height = ((2 - frame_mbs_only_flag) * self.MbHeight * 16) - self.CropTop*2 - self.CropBottom*2

Referring to the official mediacommon implementation: https://github.com/bluenviron/mediacommon/blob/main/pkg/codecs/h264/sps.go#L707-L741
